test/test_piece_picker.cpp

```diff
@@ -765,6 +778,19 @@ TORRENT_TEST(reverse_sequential_download)
 			, i % blocks_per_piece));
 }
 
+TORRENT_TEST(reverse_sequential_range_download)
+{
+	// test reverse sequential range download
+	auto p = setup_picker("7654321", "       ", "", "");
+	p->set_sequential_range(piece_index_t(1), piece_index_t(5));
+	auto picked = pick_pieces(p, "*******", 7 * blocks_per_piece, 0, nullptr
+		, piece_picker::sequential | piece_picker::reverse, empty_vector);
+	TEST_CHECK(int(picked.size()) == 5 * blocks_per_piece);
+	for (int i = 0; i < int(picked.size()); ++i)
+		TEST_CHECK(picked[std::size_t(i)] == piece_block(piece_index_t(6 - ((i + 4) / blocks_per_piece))
```

Comment:
same comment about this "4". I think it should use the named constant

Reply:
yeah again it is not blocks per piece.

Reply:
if `blocks_per_piece` changed, this literal constant would have to change exactly the same. So I think it's safe to say this is 4 because piece 1 times 4 blocks_per_piece = 4.

Reply:
Sure